### PR TITLE
Add admin module

### PR DESCRIPTION
**Pull request by ovekyc**
* Add admin router & controller
* Add admin secret
* Update authentication
* Add admin middle-ware
* Add unit tests

### CODE REVIEW

**Comment by ovekyc** [comment]
I wrote function name with underscore because they are not normal operations
The operations are like root permission op.

If you want to know how to use. Check out test code.
There is no api-doc for root permission operations

**Comment by ovekyc** [comment]
We need to make our admin secret key.
And also add it to env var

File: src/server/admin/controllers.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    db.createReadStream({
+      start: `${prefix}`,
+      end: `${prefix}\xFF`
+    })
+    .on('data', (val) => {
+      collection.push(val);
+    })
+    .on('error', (err) => {
+      res.status(500).send(err);
+      return next();
```
Comment (festinalente91) [inline src/server/admin/controllers.js]: please use APIError

File: src/server/admin/controllers.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    })
+    .on('close', () => {
+      if (res.headersSent) {
```
Comment (festinalente91) [inline src/server/admin/controllers.js]: this must have some influence by upper code

File: src/server/admin/controllers.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      next();
+      return;
+    });
+  },
+
+  db_post: (req, res, next) => {  // eslint-disable-line camelcase
+    const {key, value} = req.body;
+    db.put(key, value, (err) => {
+      if (err) {
+        res.status(500).send(err);
```
Comment (festinalente91) [inline src/server/admin/controllers.js]: ditto

File: src/server/admin/controllers.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      res.sendStatus(200);
+      return next();
+    });
+  },
+
+  db_delete: (req, res, next) => {  // eslint-disable-line camelcase
+    const key = req.params.id;
+    db.del(key, (err) => {
+      if (err) {
+        res.status(500).send(err);
```
Comment (festinalente91) [inline src/server/admin/controllers.js]: ditto

File: src/server/auth-token.js [hunk 5]
```diff
@@ -59,6 +59,9 @@ class AuthToken {
   }
 
   static authenticate(req, res, next) {
+    if (res.headersSent) {
```
Comment (festinalente91) [inline src/server/auth-token.js]: why don't you just don't use next() in the admin controller?

File: src/server/permission.js [hunk 6]
```diff
@@ -1,4 +1,7 @@
 import {APIError} from './ErrorHandler';
+import config from 'config';
+
+export const ADMIN_SECRET = process.env.ADMIN_SECRET ? process.env.ADMIN_SECRET : config.adminSecret; // eslint-disable-line max-len
```
Comment (festinalente91) [inline src/server/permission.js]: process.env.ADMIN_SECRET || config.adminSecret; seems more clear.

**Comment by festinalente91** [comment]
lgtm

**Comment by festinalente91** [comment]
lgtm lgtm

**Comment by ovekyc** [comment]
lgtm

**Comment by festinalente91** [comment]
lgtm

**Pull request merged**